Neos.ContentRepository.Core/Classes/Projection/ContentGraph/Filter/PropertyValue/PropertyValueCriteriaMatcher.php

```diff
@@ -0,0 +1,99 @@
+            default => throw new \InvalidArgumentException(sprintf('Invalid/unsupported property value criteria "%s"', get_debug_type($propertyValueCriteria)), 1679561073),
+        };
+    }
+
+    private static function propertyValueContains(PropertyCollection $propertyCollection, PropertyValueContains $propertyValueCriteria): bool
+    {
+        $propertyValue = $propertyCollection->serialized()->getProperty($propertyValueCriteria->propertyName->value)?->value;
+        if ($propertyValueCriteria->caseSensitive) {
+            return is_string($propertyValue) ? str_contains($propertyValue, $propertyValueCriteria->value) : false;
+        } else {
+            return is_string($propertyValue) ? str_contains(mb_strtolower($propertyValue), mb_strtolower($propertyValueCriteria->value)) : false;
+        }
+    }
+
+    private static function propertyValueStartsWith(PropertyCollection $propertyCollection, PropertyValueStartsWith $propertyValueCriteria): bool
+    {
+        $propertyValue = $propertyCollection->serialized()->getProperty($propertyValueCriteria->propertyName->value)?->value;
+        if ($propertyValueCriteria->caseSensitive) {
+            return is_string($propertyValue) ? str_starts_with($propertyValue, $propertyValueCriteria->value) : false;
+        } else {
+            return is_string($propertyValue) ? str_starts_with(mb_strtolower($propertyValue), mb_strtolower($propertyValueCriteria->value)) : false;
+        }
+    }
+
+    private static function propertyValueEndsWith(PropertyCollection $propertyCollection, PropertyValueEndsWith $propertyValueCriteria): bool
+    {
+        $propertyValue = $propertyCollection->serialized()->getProperty($propertyValueCriteria->propertyName->value)?->value;
+        if ($propertyValueCriteria->caseSensitive) {
+            return is_string($propertyValue) ? str_ends_with($propertyValue, $propertyValueCriteria->value) : false;
+        } else {
+            return is_string($propertyValue) ? str_ends_with(mb_strtolower($propertyValue), mb_strtolower($propertyValueCriteria->value)) : false;
+        }
+    }
+
+    private static function propertyValueEquals(PropertyCollection $propertyCollection, PropertyValueEquals $propertyValueCriteria): bool
+    {
+        if (!$propertyCollection->serialized()->propertyExists($propertyValueCriteria->propertyName->value)) {
+            return false;
+        }
+        $propertyValue = $propertyCollection->serialized()->getProperty($propertyValueCriteria->propertyName->value)?->value;
+        if ($propertyValueCriteria->caseSensitive) {
+            return $propertyValue == $propertyValueCriteria->value;
+        }
+        return (is_string($propertyValue) ? mb_strtolower($propertyValue) : $propertyValue) == (is_string($propertyValueCriteria->value) ? mb_strtolower($propertyValueCriteria->value) : $propertyValueCriteria->value);
```

Comment:
```suggestion
            return $propertyValue === $propertyValueCriteria->value;
        }
        return (is_string($propertyValue) ? mb_strtolower($propertyValue) : $propertyValue) === (is_string($propertyValueCriteria->value) ? mb_strtolower($propertyValueCriteria->value) : $propertyValueCriteria->value);
```

To fix https://github.com/neos/neos-development-collection/pull/4701#issuecomment-1949439865

Reply:
Adjusted as requested and backed by a test